diff --git a/node_modules/@oclif/plugin-legacy/lib/hooks/init.js b/node_modules/@oclif/plugin-legacy/lib/hooks/init.js
index 4e8addd..60fa4dd 100644
--- a/node_modules/@oclif/plugin-legacy/lib/hooks/init.js
+++ b/node_modules/@oclif/plugin-legacy/lib/hooks/init.js
@@ -5,7 +5,7 @@ const util_1 = require("../util");
 exports.init = async function (opts) {
     const plugins = await Promise.all(opts.config.plugins.map(async (p, i) => {
         if (p.valid)
-            return;
+            return p;
         try {
             const plugin = new __1.PluginLegacy(opts.config, p);
             await plugin.load();
